src/Extensions/GoogleSitemapSiteTreeExtension.php

```diff
@@ -160,7 +160,7 @@ public function ImagesForSitemap()
             }
         }
 
-        $this->owner->extend('updateImagesForSitemap', $list);
+        $list = $this->owner->extend('updateImagesForSitemap', $list);
```

Comment:
List is an ArrayList so it should be passed as a reference. You don't need to change this - just  call  $list->removeAll()

Reply:
Hi Wilr, $list->removeAll() is not a function in ArrayList, throws an error instead... But that's all good, I'll find another way to get this working

Reply:
`    public function updateImagesForSitemap($list)
    {
        foreach ($list as $item) {
            $list->remove($item);
        }
    }
`

Reply:
That does it